crud/src/main/java/com/redhat/lightblue/assoc/ep/Copy.java

```diff
@@ -0,0 +1,63 @@
+/*
+ Copyright 2013 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.assoc.ep;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+
+/**
+ * Copies the results from another step
+ */
+public class Copy extends AbstractSearchStep {
+
+    private final Source<ResultDocument> source;
+    
+    public Copy(ExecutionBlock block,Source<ResultDocument> source) {
+        super(block);
+        this.source=source;
+    }
+
+    @Override
+    public StepResult<ResultDocument> getResults(ExecutionContext ctx) {
+        return new StepResultWrapper<ResultDocument>(source.getStep().getResults(ctx)) {
+            @Override
+            public Stream<ResultDocument> stream() {
+                // Create new documents for each document in the source. This will
+                // create the correct slots based on this execution block
+                return super.stream().map(d->new ResultDocument(block,d.getDoc()));
+            }
+        };
+    }
+
+    @Override
+    protected final List<ResultDocument> getSearchResults(ExecutionContext ctx) {
+        return null;
```

Comment:
Throwing an exception instead. This shouldn't be called

On Tue, Mar 29, 2016 at 4:23 PM, Brandon Vulaj <notifications@github.com>
wrote:

> In crud/src/main/java/com/redhat/lightblue/assoc/ep/Copy.java
> <https://github.com/lightblue-platform/lightblue-core/pull/608#discussion_r57811172>
> :
>
> > +
> > +    @Override
> > +    public StepResult<ResultDocument> getResults(ExecutionContext ctx) {
> > +        return new StepResultWrapper<ResultDocument>(source.getStep().getResults(ctx)) {
> > +            @Override
> > +            public Stream<ResultDocument> stream() {
> > +                // Create new documents for each document in the source. This will
> > +                // create the correct slots based on this execution block
> > +                return super.stream().map(d->new ResultDocument(block,d.getDoc()));
> > +            }
> > +        };
> > +    }
> > +
> > +    @Override
> > +    protected final List<ResultDocument> getSearchResults(ExecutionContext ctx) {
> > +        return null;
>
> Can we return a n empty list here instead?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/608/files/d72d06258b9b58a53294c8eb4a994145df771c4c#r57811172>
>